### PR TITLE
Add cifar10

### PR DESCRIPTION
**Pull request by Chen-Roth**

### CODE REVIEW

File: examples/cifar10/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+profile perspective.
+
+![image_1](screen_shots_2/cluset_bisecting_number_12/Screenshot 2023-07-13 at 11.04.56.png)
+![image_2](screen_shots_2/cluset_bisecting_number_12/Screenshot 2023-07-13 at 11.05.09.png)
+![image_3](screen_shots_2/cluset_bisecting_number_12/Screenshot 2023-07-13 at 11.05.23.png)
+
+## Sample Loss Analysis
+
+Within this section, we delve into the examination of features that impact the model's predictions. Through the 
+identification and extraction of the most influential features, as well as assessing the derivatives of the loss
```
Comment (tomkoren21) [inline examples/cifar10/README.md:63]: influential -> prominent

Comment (tomkoren21) [reply examples/cifar10/README.md:63]: I would also not explicitly say derivates of the loss function in relation to activation - this could be removed. Maybe something like:
"TL automatically generates a heatmap that quantifies the effect different features has on the loss.. In the image depicated below...."

File: examples/cifar10/cifar10/data_set.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from keras.utils import to_categorical
+from sklearn.model_selection import train_test_split
+
+LABELS = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']
+LABELS_NAMES = ['airplane', 'automobile', 'bird', 'cat', 'deer', 'dog', 'frog', 'horse', 'ship', 'truck']
+
+
+# Preprocess Function
+def preprocess_func():
+    (data_X, data_Y), (_, _) = cifar10.load_data()
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:13]: might be cleaner to do this if poosible:
(data_X, data_Y), _ = cifar10.load_data()

File: examples/cifar10/cifar10/data_set.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    data_Y = to_categorical(data_Y)  # Hot Vector
+
+    train_X, val_X, train_Y, val_Y = train_test_split(data_X, data_Y, test_size=0.2, random_state=42)
+
+    return train_X, val_X, train_Y, val_Y
+
+
+# Metadata functions allow to add extra data for a later use in analysis.
+# This metadata adds the int gt_digit of each sample (not a hot vector).
+def metadata_gt_label(one_hot_digit) -> int:
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:26]: add input param type

File: examples/cifar10/cifar10/data_set.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    digit = one_hot_digit.argmax()  # Returns the indices of the maximum values along an axis
+    digit_int = int(digit)
+    return digit_int
+
+# This metadata adds the int gt_name of each sample.
+def metadata_label_name(one_hot_digit) -> str:
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:32]: add input param type

File: examples/cifar10/cifar10/data_set.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    digit = one_hot_digit.argmax()  # Returns the indices of the maximum values along an axis
+    digit_int = int(digit)
+    return LABELS_NAMES[digit_int]
+
+
+# This metadata adds each sample if it is contains to airplane or bird.
+def metadata_fly(one_hot_digit) -> str:
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:39]: add input param type

File: examples/cifar10/cifar10/data_set.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    digit = one_hot_digit.argmax()  # Returns the indices of the maximum values along an axis
+    digit_int = int(digit)
+    if digit_int in [0, 2]:
+        return 'airplane or bird'
+    return 'else'
+
+
+
+# This metadata adds each sample if it is vehicle ar animal.
+def metadata_animal(one_hot_digit) -> str:
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:49]: add input param type

File: examples/cifar10/cifar10/data_set.py
Comment (tomkoren21) [inline examples/cifar10/cifar10/data_set.py:1]: Consider renaming to helper_functions/utils or something else

File: examples/cifar10/cifar10/encoders.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+
+import numpy as np
+from scipy.ndimage import zoom
+
+# Input encoder fetches the image with the index `idx` from the `images` array set in
+# the PreprocessResponse data. Returns a numpy array containing the sample's image.
+def input_encoder(image) -> np.ndarray:
```
Comment (tomkoren21) [inline examples/cifar10/cifar10/encoders.py:7]: type of param

**Review by tomkoren21 — APPROVED** [review]
LGTM other than small changes
